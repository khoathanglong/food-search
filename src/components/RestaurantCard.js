import { PromotionIcon } from "./PromotionIcon";
import { Star } from "@mui/icons-material";
import { Card, CardMedia, CardContent, Typography, Chip } from "@mui/material";
import { useMemo } from "react";

export const RestaurantCard = ({
  imageUrl,
  rating,
  promotion,
  isNew,
  minCookTime,
  maxCookTime,
  restaurant,
}) => {
  const calculatedRating = useMemo(
    () => Math.round((rating + Number.EPSILON) * 10) / 10,
    [rating]
  );
  return (
    <Card sx={{ position: "relative", borderRadius: 2 }} elevation={2}>
      <CardMedia
        component="img"
        height="140"
        image={imageUrl}
        alt={restaurant}
      />
      <PromotionIcon promotion={promotion} />
      <CardContent>
        <Typography
          gutterBottom
          variant="body1"
          component="div"
          sx={{ fontWeight: "bold", mb: 2 }}
        >
          {restaurant}
        </Typography>
        <Chip
          size="small"
          icon={<Star />}
          label={calculatedRating}
          sx={{ borderRadius: 2 }}
        />
        <Chip
          size="small"
          label={`${minCookTime}-${maxCookTime} min`}
          sx={{ mx: 1, borderRadius: 2 }}
        />
        {isNew && (
          <Chip
            size="small"
            label="New"
            sx={{ color: "success.light", borderRadius: 2, fontWeight: "bold" }}
          />
        )}
      </CardContent>
    </Card>
  );
};
