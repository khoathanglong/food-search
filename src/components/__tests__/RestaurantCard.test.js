import { render } from "@testing-library/react";
import { RestaurantCard } from "../RestaurantCard";

describe("RestaurantCard", () => {
  it("renders correctly when isNew=false", () => {
    const mockProps = {
      rating: 4.6886,
      promotion: "1+1",
      isNew: false,
      minCookTime: 100,
      maxCookTime: 120,
      restaurant: "Kog",
      imageUrl: "https://source.unsplash.com/random/400x400?Desserts",
    };
    const { container } = render(<RestaurantCard {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
  it("renders correctly when isNew=true", () => {
    const mockProps = {
      rating: 4.6886,
      promotion: "1+1",
      isNew: true,
      minCookTime: 100,
      maxCookTime: 120,
      restaurant: "Kog",
      imageUrl: "https://source.unsplash.com/random/400x400?Desserts",
    };
    const { container } = render(<RestaurantCard {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
