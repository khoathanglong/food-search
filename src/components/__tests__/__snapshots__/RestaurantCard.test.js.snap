// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard renders correctly when isNew=false 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-qoz6c7-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Kog"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="https://source.unsplash.com/random/400x400?Desserts"
    />
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-153d81e-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
      >
        1+1
      </span>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-3dl3ms-MuiTypography-root"
      >
        Kog
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1tsv6p9-MuiChip-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          4.7
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-cnx0r8-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          100-120 min
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`RestaurantCard renders correctly when isNew=true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-qoz6c7-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Kog"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="https://source.unsplash.com/random/400x400?Desserts"
    />
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-153d81e-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
      >
        1+1
      </span>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-3dl3ms-MuiTypography-root"
      >
        Kog
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1tsv6p9-MuiChip-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          4.7
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-cnx0r8-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          100-120 min
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-16ku107-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          New
        </span>
      </div>
    </div>
  </div>
</div>
`;
